job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -514,10 +515,47 @@ class WebApi(system: ActorSystem,
    *    GET /healthz              - return OK or error message
    */
   def healthzRoutes: Route = pathPrefix("healthz") {
+    import ContextSupervisor._
+    import JobManagerActor._
+    import JobInfoActor._
+    import JobDAOActor._
     //no authentication required
     get { ctx =>
       logger.info("Receiving healthz check request")
-      ctx.complete("OK")
+      var actorsAlive: Int = 0
+      val supervisorFuture = (supervisor ? GetContext("dummycontext")).mapTo[NoSuchContext.type]
+      val jobInfofuture = (jobInfoActor ? GetJobStatus("dummyjobid")).mapTo[NoSuchJobId.type]
+      val jobDaofuture = (daoActor ? GetJobInfos(1)).mapTo[JobInfos]
+      val statusActor = system.actorOf(JobStatusActor.props(daoActor))
+      val jobStatusfuture = (statusActor ? Unsubscribe("dummyjobid", null)).mapTo[NoSuchJobId.type]
+      val resultActor = GetResultActor("dummycontext")
+      val jobResultfuture = (jobInfoActor ? GetJobResult("dummyjobid")).mapTo[NoSuchJobId.type]
+
+      val listOfFutures = Seq(supervisorFuture, jobInfofuture, jobDaofuture, jobStatusfuture, jobResultfuture)
+      val futureOfList = Future.sequence(listOfFutures)
+
+      try {
+        val results = Await.result(futureOfList, 60 seconds)
+        for (result <- results) {
+          result match {
+            case NoSuchContext => actorsAlive += 1
+            case NoSuchJobId => actorsAlive += 1
+            case JobInfos(jobInfos) => actorsAlive += 1
+            case _ => logger.error("Unexpected response!")
+         }
+        }
+      } catch {
+        case ex: Exception => logger.error(ex.getMessage())
+      }
+
+      if (actorsAlive == 5) {
+        logger.info("Required actors alive")
+        ctx.complete(StatusCodes.OK)
+      }
+      else {
+        logger.error("Required actors not alive")
+        ctx.complete(500, errMap("Required actors not alive"))
```

Comment:
Just as a suggestion: Maybe after removing a check for some of the actors above, you can also easier track which ones are not here.  Otherwise you will need to waste time debugging which actor is not present.

Reply:
I shall take up this suggestion as an enhancement to this API.